src/picomsg/src/routing_table/RoutingTable.cpp

```diff
@@ -0,0 +1,104 @@
+
+#include "RoutingTable.hpp"
+#include "logger.h"
+
+namespace picomsg {
+
+namespace detail {
+
+/*!*!*!*!*!*!*!
+ * RouteSPKey *
+ *!*!*!*!*!*!*/
+RouteSPKey::RouteSPKey(const struct routing_table_route_entry_sp_info& route_entry_sp_info)
+	: m_route_entry_sp_info(route_entry_sp_info)
+{ }
+
+/*!*!*!*!*!*!*!
+ * RouteEntry *
+ *!*!*!*!*!*!*/
+RouteEntry::RouteEntry(const struct routing_table_route_entry& route_entry)
+	: m_route_entry(route_entry)
+{ }
+
+const struct routing_table_route_entry& RouteEntry::to_raw() const {
+	return m_route_entry;
+}
+struct routing_table_route_entry& RouteEntry::to_raw() {
+	return m_route_entry;
+}
+
+} /* namespace detail { */
+
+RoutingTable::RoutingTable() {
+}
+
+RoutingTable::~RoutingTable() {
+}
+
+bool RoutingTable::init() {
+	return true;
+}
+
+std::unique_ptr<RoutingTable> RoutingTable::create() {
+	// Thank you C++
+	std::unique_ptr<RoutingTable> table { new(std::nothrow) RoutingTable{} };
```

Comment:
I'd rather use `make_unique` for exception safety. Anyway `new` will rarely throw.

Reply:
`make_unique` doens't compile since the constructor is private

Reply:
why ctor private then?

Reply:
Forces users to call `create()` as the only way of creating an object, to avoid missing calling `init()` after constructing an object.